jetty-websocket/websocket-common/src/main/java/org/eclipse/jetty/websocket/common/events/JettyAnnotatedEventDriver.java

```diff
@@ -152,6 +152,15 @@ public void onConnect()
         }
     }
 
+    @Override
+    public void onContinuationFrame(ByteBuffer buffer, boolean fin) throws IOException
+    {
+        if (events.onText != null || events.onBinary != null)
+        {
+            super.onContinuationFrame(buffer, fin);
```

Comment:
I think this will still fail if there's only a `@OnWebSocketMessage` method for text messages and a continuation frame for a binary message is received or vice versa.

Reply:
yeah, I considered that.
the existing testcases that only declare `@OnWebSocketMessage` were happy with this change.

Reply:
See testcase in commit 02c49efc5f5c9768ba5ad15349fe9565ba7333e0

It uses the Endpoint defined as ...

https://github.com/eclipse/jetty.project/blob/02c49efc5f5c9768ba5ad15349fe9565ba7333e0/jetty-websocket/jetty-websocket-tests/src/test/java/org/eclipse/jetty/websocket/tests/EchoSocket.java#L28-L36

Reply:
I went with a different approach.
See commit f36358c631cf24fcf2797eec4ee33760b76fa60d